src/data/finset/lattice.lean

```diff
@@ -318,6 +332,12 @@ begin
       { exact ⟨b, mem_cons.2 (or.inr hb), sup_eq_right.2 h⟩, }, }, },
 end
 
+lemma sup'_mem
+  (s : set α) (w : ∀ x y ∈ s, x ⊔ y ∈ s)
+  {ι : Type*} (t : finset ι) (H : t.nonempty) (p : ι → α) (h : ∀ i, p i ∈ s) :
+  t.sup' H p ∈ s :=
+sup'_induction H p w (λ i _, h i)
```

Comment:
We may as well weaken the hypothesis, right?
```suggestion
  {ι : Type*} (t : finset ι) (H : t.nonempty) (p : ι → α) (h : ∀ i ∈ t, p i ∈ s) :
  t.sup' H p ∈ s :=
sup'_induction H p w h
```

Reply:
Yes.